### PR TITLE
PPTP-1787 Update with deregistration details and call to NRS

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed

### CODE REVIEW

**Review by tonymccrae-ee — APPROVED** [review]

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/DeregistrationControllerSpec.scala [hunk 1]
```diff
@@ -64,16 +91,95 @@ class DeregistrationControllerSpec extends ControllerSpec {
       deregistrationDetails.deregistrationDeclarationBox1 mustBe true
     }
 
+    "EIS/IF subscription is successful" when {
+      " NRS submission is successful" in {
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/DeregistrationControllerSpec.scala:95]: It's not obvious what this test is trying to prove based on the name.

Comment (RyanKellyHMRC) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/DeregistrationControllerSpec.scala:95]: Perhaps `return expected deregistration response details` when `NRS submission is successful` is better?

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/DeregistrationControllerSpec.scala [hunk 2]
```diff
@@ -64,16 +91,95 @@ class DeregistrationControllerSpec extends ControllerSpec {
...
+        )
+        verify(mockNonRepudiationService).submitNonRepudiation(
+          ArgumentMatchers.eq(toJson(updatedSubscription).toString()),
+          any[ZonedDateTime],
+          ArgumentMatchers.eq(subscriptionSuccessfulResponse.pptReferenceNumber),
+          any()
+        )(any[HeaderCarrier])
+      }
+
+      " NRS submission fails with failure response" in {
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/DeregistrationControllerSpec.scala:133]: And `return expected deregistration response details` when `NRS submission fails` here?

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/models/DeregistrationReasonSpec.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          "Below De-minimus"
+        ) mustBe DeregistrationReason.NotMetAndDoNotExpectToMeetThreshold
+      }
+    }
+
+    "throw IllegalStateExceptions" when {
+      "key is absent" when {
+        "accessing deregistration reason" in {
+          intercept[IllegalStateException] {
+            DeregistrationReason.apply("Below de-minimus")
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/models/DeregistrationReasonSpec.scala:51]: That's a subtle incorrect value. Perhaps use something more obviously wrong like `XXX`?

**Review by RyanKellyHMRC — APPROVED** [review]
Looks good. Just a couple of minor observations around test names and test data.

**Pull request merged**